File: InterviewBit/fizz_buzz.py
# https://www.interviewbit.com/problems/fizzbuzz/
'''
	Author - Subhajit Das
	University of Engineering and Management, Kolkata
	31/1/2018
'''


class Solution:
    # @param A : integer
    # @return a list of strings
    def fizzBuzz(self, n):
        arr = list(range(n+1))

        for idx in range(1, n+1):
            if idx % 3 == 0 and idx % 5 == 0:
                arr[idx] = 'FizzBuzz'
            elif idx % 3 == 0:
                arr[idx] = 'Fizz'
            elif idx % 5 == 0:
                arr[idx] = 'Buzz'
            else:
                arr[idx] == str(idx)
        
        arr.pop(0)
        return arr

File: Leetcode/fizz_buzz.java
// https://leetcode.com/problems/fizz-buzz/
/*
    Author @Subhajit Das (sdsubhajitdas.github.io)
    SWE @Turbot HQ India PVT Ltd.
    29/11/2020
*/
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> fizzBuzz(int n) {

        List<String> fizzBuzzList = new ArrayList<>();
        for(int i=1; i<=n;i++){
            fizzBuzzList.add(helperFizzBuzz(i));
        }
        return fizzBuzzList;
        
    }

    private String helperFizzBuzz(int n){
        String fizzBuzz = "";
        if(n % 3 == 0)
            fizzBuzz += "Fizz";
        if(n % 5 == 0)
            fizzBuzz += "Buzz";
        return (fizzBuzz == "")? Integer.toString(n) :fizzBuzz;
    }
}